### PR TITLE
Make PluginConfigOption generic

### PR DESCRIPTION
**Pull request by echlebek**
This commit refactors the PluginConfigOption and the things surrounding
    it so that PluginConfigOption now has a type parameter, and can
    therefore express concrete types instead of interface{} for its Value
    and Default struct fields.
    
    Additionally, some whole-project maintenance has been done, including
    upgrading the core/v2 module, refactoring tests somewhat, and
    refactoring goplugin.go.
    
    Signed-off-by: Eric Chlebek <eric@sensu.io>

### CODE REVIEW

File: sensu/goplugin.go [hunk 1]
```diff
@@ -146,76 +169,56 @@ func (p *basePlugin) initPlugin() error {
...
+		if err := opt.SetupFlag(cmd); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func setupFlag(cmd *cobra.Command, opt *PluginConfigOption) error {
-	if len(opt.Argument) == 0 {
+func (p *PluginConfigOption[T]) SetupFlag(cmd *cobra.Command) error {
```
Comment (c-kruse) [inline sensu/goplugin.go:179]: Looks like Default values may be have been dropped. Probably makes sense for them to get set here.

**Review by c-kruse — CHANGES_REQUESTED** [review]
Really excited for this!

I just started tinkering with it but have ran into a one gotcha.

File: sensu/goplugin.go (hunk 1 above)
Comment (echlebek) [reply sensu/goplugin.go:179]: Good call, I missed porting that case over. Surprised the tests didn't catch it, I'll add some to make sure it's covered.

**Review by c-kruse — COMMENTED** [review]
Lack of default values was the only regression I could find poking around!

Pretty straight-forward to upgrade to and use.

**Comment by echlebek** [comment]
There wasn't a straightforward way to test that I'm calling viper.SetDefault, so I've left it untested.

**Review by c-kruse — APPROVED** [review]

**Comment by echlebek** [comment]
Merging this now. This is going to break all of the existing plugins, so I'm going to follow it up with some more breaking changes.

**Pull request merged**